### PR TITLE
Create appointment step two

### PR DESCRIPTION
**Pull request by LomyW**
no real data from BE

### CODE REVIEW

File: src/widgets/appointment/AppointmentList.tsx [hunk 1]
```diff
@@ -1,40 +1,19 @@
...
-//             id
-//         },
-//         isCancelled,
-//         appointmentType
-//   }
-// }
-// `);
+type ListProps = {
+    appointments?: any;
+    isStatic: boolean;
```
Comment (LomyW) [inline src/widgets/appointment/AppointmentList.tsx:9]: open for name suggestions

File: src/pages/create-appointment/AppointmentsView.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import { useLayoutHelper } from '../../hooks/useLayoutHelper';
+import AppointmentList from '../../widgets/appointment/AppointmentList';
+
+type Props = {
+    courseId?: number;
+    next: () => void;
+    back: () => void;
+};
+
+// query myappointments {
```
Comment (LomyW) [inline src/pages/create-appointment/AppointmentsView.tsx:14]: this is a example query, will may change

File: src/components/appointment/AppointmentDetail.tsx [hunk 3]
```diff
@@ -19,6 +19,15 @@ const AppointmentDetail: React.FC<AppointmentDetailProps> = ({ appointment, cour
     const toast = useToast();
     const { space, sizes } = useTheme();
     const [canceled, setCanceled] = useState<boolean>(false);
+    const { date, startTime, endTime } = getAppointmentDateTime(appointment.start, appointment.duration);
+
+    const countAttendees = () => {
```
Comment (timidak) [inline src/components/appointment/AppointmentDetail.tsx:24]: This function is created on each render, you could refactor this with `useCallback`.

File: src/components/appointment/AppointmentDetail.tsx [hunk 4]
```diff
@@ -19,6 +19,15 @@ const AppointmentDetail: React.FC<AppointmentDetailProps> = ({ appointment, cour
...
+        // TODO change to appointment.appointment_participant_pupil, appointment.appointment_participant_student, appointment.appointment_participant_screener
+        const participants = appointment.participants ? appointment.participants.length : 0;
+        const organizers = appointment.organizers ? appointment.organizers.length : 0;
+        return participants + organizers;
+    };
+    const attendeesCount = countAttendees();
```
Comment (timidak) [inline src/components/appointment/AppointmentDetail.tsx:30]: Here `useMemo` could help

File: src/components/appointment/Description.tsx [hunk 5]
```diff
@@ -3,7 +3,7 @@ import { useTranslation } from 'react-i18next';
 import { AppointmentTypes } from '../../types/lernfair/Appointment';
 
 type AvatarsProps = {
-    appointmentType: string;
+    appointmentType?: string;
```
Comment (timidak) [inline src/components/appointment/Description.tsx:6]: I'm not sure if a `string` is a valid type for "Type". I would rather use a union or enum.

File: src/components/appointment/Header.tsx [hunk 6]
```diff
@@ -3,8 +3,8 @@ import { useTranslation } from 'react-i18next';
 import { AppointmentTypes, Attendee } from '../../types/lernfair/Appointment';
 
 type HeaderProps = {
-    appointmentType: string;
-    organizers: Attendee[];
+    appointmentType?: string;
```
Comment (timidak) [inline src/components/appointment/Header.tsx:6]: > I'm not sure if a string is a valid type for "Type". I would rather use a union or enum.

Same here

File: src/pages/CreateAppointment.tsx [hunk 7]
```diff
@@ -1,32 +1,45 @@
...
     const onBack = useCallback(() => {
-        if (currentIndex === 0) return;
         setCurrentIndex((prev) => prev - 1);
     }, []);
 
+    const skipStepTwo = useCallback(() => {
+        setCurrentIndex((prev) => prev + 2);
+    }, [currentIndex]);
+
+    const onStepOne = (id?: number) => {
```
Comment (timidak) [inline src/pages/CreateAppointment.tsx:26]: Naming sounds off in my opinion.

**Review by timidak — CHANGES_REQUESTED** [review]
Some minor changes 👍🏻

File: src/pages/CreateAppointment.tsx [hunk 8]
```diff
@@ -1,32 +1,45 @@
...
-                    <InstructionProgress
-                        currentIndex={currentIndex}
-                        instructions={[{ label: 'Zuordnung wählen' }, { label: 'Termine einsehen' }, { label: 'Termin hinzufügen' }]}
-                    />
-                    {currentIndex === 0 && <AppointmentAssignment next={onNext} back={onBack} />}
+                <Box mx="4">
+                    <View position="sticky" mb={2} overflow="hidden">
+                        <InstructionProgress
+                            currentIndex={currentIndex}
+                            instructions={[{ label: 'Zuordnung wählen' }, { label: 'Termine einsehen' }, { label: 'Termin hinzufügen' }]}
```
Comment (timidak) [inline src/pages/CreateAppointment.tsx:38]: Labels should be added to i18n

File: src/pages/CreateAppointment.tsx [hunk 9]
```diff
@@ -1,32 +1,45 @@
...
     }, []);
 
+    const skipStepTwo = useCallback(() => {
+        setCurrentIndex((prev) => prev + 2);
+    }, [currentIndex]);
+
+    const goToStepTwo = (id?: number) => {
+        if (id) setCourseId(id);
+        onNext();
+    };
```
Comment (timidak) [inline src/pages/CreateAppointment.tsx:29]: You could wrap this in a useCallback to I guess. 
```typescript
 const goToStepTwo = useCallback((id?: number) => {
        if (id) setCourseId(id);
        onNext();
    }, [setCourseId, onNext]);
```

File: src/pages/create-appointment/AppointmentsView.tsx (hunk 2 above)
Comment (timidak) [reply src/pages/create-appointment/AppointmentsView.tsx:14]: What about this query now? Can we remove it?

File: src/widgets/appointment/AppointmentList.tsx (hunk 1 above)
Comment (timidak) [reply src/widgets/appointment/AppointmentList.tsx:9]: isReadOnly

File: src/widgets/appointment/AssignmentTile.tsx [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+                            <Box h="100%" w={containerWidth} padding={space['0.5']}>
+                                <Image
+                                    position="absolute"
+                                    left={0}
+                                    right={0}
+                                    top={0}
+                                    width="100%"
+                                    height="100%"
+                                    bg="gray.400"
+                                    alt={'Kursbild'}
```
Comment (timidak) [inline src/widgets/appointment/AssignmentTile.tsx:64]: For SEO and Accsessibility reasons, we should translate alt text to the language of the user. So, I would add this to i18n.

File: src/widgets/appointment/appointment-list/CalendarWeek.tsx [hunk 11]
```diff
@@ -1,17 +1,18 @@
...
 
 type WeekProps = {
     key: React.Key;
     appointmentsOfWeek: AppointmentType[];
     lastOfMonth: boolean;
-    scrollToRef: any;
-    scrollId: number;
+    scrollToRef?: any;
+    scrollId?: number;
+    isStatic?: boolean;
```
Comment (timidak) [inline src/widgets/appointment/appointment-list/CalendarWeek.tsx:12]: isReadOnly 😸

**Review by timidak — APPROVED** [review]
Some last changes 👍🏻 
Rest looks pretty good!

Just merge it, when you added the changes.

File: src/pages/CreateAppointment.tsx (hunk 8 above)
Comment (LomyW) [reply src/pages/CreateAppointment.tsx:38]: is done in PR #128

**Pull request merged**